test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```diff
@@ -448,6 +448,35 @@
               "No exception thrown even though no file existed for copying")
           (fs/copy copy cacert))))))
 
+(deftest retrieve-ca-crl!-test
+  (testing "CRL file copied when it doesn't already exist"
+    (let [tmp-confdir (fs/copy-dir confdir (ks/temp-dir))
```

Comment:
This is a common pattern we have with the CA tests - there's usually a tmp dir "sandbox" that the subsequent `testing` blocks operate within.  If each `testing` block was executed concurrently then yes, we would definitely have problems, but we don't run tests that way.  I would think that parallelization would happen at the `deftest` level rather than the lower `testing` level though, so we shouldn't have that problem.

Or maybe I'm wrong and you can parallelize the individual `testing` blocks?  If we decided to do that we'd have some other tests to redo anyways.  Either way, we're planning on doing some heavy refactoring with the persistence of the CA data so this filesystem test code won't live much longer.